contributor_docs/documentation_style_guide.md

```diff
@@ -0,0 +1,1184 @@
+
+```javascript
+// Bad.
+if (mouseIsPressed) {
+  // ...
+}
+
+// Good.
+if (mouseIsPressed === true) {
+  // ...
+}
+
+// Bad.
+if (name) {
+  // ...
+}
+
+// Good.
+if (name !== '') {
+  // ...
+}
+
+// Bad.
+if (collection.length) {
+  // ...
+}
+
+// Good.
+if (collection.length > 0) {
+  // ...
+}
+```
+
+- Don't use `switch` statements unless it's necessary.
+
+- Use parentheses when mixing operators. The only exceptions are the arithmetic operators `+`, `-`, and `**`.
+
+> Why? It's easier to read and avoids subtle bugs.
+
+```javascript
+// Bad.
+const huh = a && b < 0 || c > 0 || d + 1 === 0;
+
+// Good.
+const huh = (a && b < 0) || c > 0 || (d + 1 === 0);
+
+// Bad.
+if (a || b && c) {
+  return d;
+}
+
+// Good.
+if (a || (b && c)) {
+  return d;
+}
+
+// Bad.
+const what = a + b / c * d;
+
+// Good.
+const what = a + (b / c) * d;
+```
+
+## Conditionals
+
+- Use braces with all multiline blocks.
+
+```javascript
+// Bad.
+if (mouseIsPressed === true)
+  circle(mouseX, mouseY, 50);
+
+// Better.
+if (mouseIsPressed === true) circle(mouseX, mouseY, 50);
+
+// Best.
+if (mouseIsPressed === true) {
+  circle(mouseX, mouseY, 50);
+}
+```
+
+- Put `else` on the same line as the preceding `if` block’s closing brace.
+
+```javascript
+// Bad.
+if (mouseIsPressed === true) {
+  thing1();
+  thing2();
+}
+else {
+  thing3();
+}
+
+// Good.
+if (mouseIsPressed === true) {
+  thing1();
+  thing2();
+} else {
+  thing3();
+}
+```
+
+- Don't use an `else` block after an `if` block that always executes a `return` statement.
+
+```javascript
+// Bad.
+function mouseIsOnLeft() {
+  if (mouseX < width * 0.5) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+function mouseIsOnLeft() {
+  if (mouseX < width * 0.5) {
+    return true;
+  }
+
+  return false;
+}
+```
+
+- If a condition gets too long, place each (grouped) condition on a new line. The logical operator should begin the line.
+
+> Why? It's easier to read.
+
+```javascript
+// Bad.
+if ((number === 123 || letters === 'abc') && mouseIsPressed === true && keyIsPressed === true) {
+  doStuff();
+}
+
+// Good.
+if (
+  (number === 123 || letters === 'abc')
+  && mouseIsPressed === true
+  && keyIsPressed === true
+) {
+  doStuff();
+}
+```
+
+- Don't use selection operators in place of conditionals.
+
+  ```javascript
+  // Bad.
+  refrigeratorIsRunning && goCatchIt();
+
+  // Good.
+  if (refrigeratorIsRunning === true) {
+    goCatchIt();
+  }
+  ```
+
+**[⬆ back to top](#table-of-contents)**
+
+## Iteration
+
+- Don’t use a `while` or `do-while` loops unless it's necessary. Use `for` loops to iterate a fixed number of times.
+
+```javascript
+const numPetals = 7;
+
+// Bad.
+let i = 0;
+while (i < numPetals) {
+  ellipse(0, 0, 20, 80);
+  rotate(PI / numPetals);
+  i += 1;
+}
+
+// Good.
+for (let i = 0; i < numPetals; i += 1) {
+  ellipse(0, 0, 20, 80);
+  rotate(PI / numPetals);
+}
+```
+
+- Don’t use `for` loops to iterate over arrays.
```

Comment:
I agree with the reasoning with `forEach()` being generally easier to read than a C-style `for`, although it's maybe worth noting that in [the learn docs for program flow](https://p5js.org/learn/program-flow.html) we don't mention any array functions for looping. We might want to add to that so users have a reference for the structures they see in the docs.

I think we also don't have many arrow functions in the docs currently, so we probably would want to explain those a bit more somewhere if we're going to use them for all looping constructs.

Reply:
Great suggestion. I'll weave this into my work over the coming months. We'll probably need a similar update to the [Foundation](https://p5js.org/reference/#group-Foundation) section of the Reference page.